skopt/utils.py

```diff
@@ -44,3 +51,70 @@ def create_result(Xi, yi, space=None, rng=None, specs=None, models=None):
     res.random_state = rng
     res.specs = specs
     return res
+
+
+def dump(res, filename, store_objective=True, **kwargs):
+    """
+    Store an skopt optimization result into a file.
+
+    Parameters
+    ----------
+    * `res` [`OptimizeResult`, scipy object]:
+        Optimization result object to be stored.
+
+    * `filename` [string or `pathlib.Path`]:
+        The path of the file in which it is to be stored. The compression
+        method corresponding to one of the supported filename extensions ('.z',
+        '.gz', '.bz2', '.xz' or '.lzma') will be used automatically.
+
+    * `store_objective` [boolean, default=True]:
+        Whether the objective function should be stored. Set `store_objective`
+        to `False` if your objective function (`.specs['args']['func']`) is
+        unserializable (i.e. if an exception is raised when trying to serialize
+        the optimization result). Notice that in this case a deep copy of the
+        optimization result is created. This can potentially lead to performance
+        problems.
```

Comment:
You can add the note that it is advisable to delete the objective function before dumping to avoid such issues